src/workers/cdc/primary.ts

```diff
@@ -795,12 +795,42 @@ const onSquadPublicRequestChange = async (
   });
 };
 
+const setRestoreStreakCache = async (
+  userId: string,
+  lastViewAt: Date,
+  previousStreak: number,
+) => {
+  const today = new Date();
+  const lastViewDifference = differenceInDays(today, lastViewAt);
+
+  if (lastViewDifference !== 2) {
```

Comment:
- It will never be less than 2 as we don't restore when it has only been a day since user can still read something for the streak today.
- When it is more than 2 days now, it means the user is not eligible anymore to restore it. We only allow it on the day you lose the streak (until the day ends).